src/controllers/blocks/BlocksController.ts

```diff
@@ -215,15 +217,15 @@ export default class BlocksController extends AbstractController<BlocksService>
 	 */
 	private parseFinalizationOpts = async (
 		optFinalizes: boolean,
-		paramFinalized: boolean
+		paramFinalized: string
 	): Promise<IFinalizationOpts> => {
 		let hash, queryFinalizedHead, omitFinalizedTag;
 		if (!optFinalizes) {
 			// If the network chain doesn't finalize blocks, we dont want a finalized tag.
 			omitFinalizedTag = true;
 			queryFinalizedHead = false;
 			hash = (await this.api.rpc.chain.getHeader()).hash;
-		} else if (!paramFinalized) {
+		} else if (paramFinalized === 'false') {
```

Comment:
oh, this check is changed just because it is a `String` now?

Reply:
So initially before i abstracted this logic out it was always a string. Then I abstracted out of the main controller logic, and wanted to get it more organized, which made the logic more convoluted and confusing which I think is what introduced the mix in logic. So for simplicity sake, and making sure it works, I am reverting it back to what it was originally.